### PR TITLE
Papyrus

### PR DESCRIPTION
**Pull request by itsbazke**

### CODE REVIEW

File: src/main/resources/assets/ltextras/models/block/papyrus_stem.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    "elements": [
+        {   "from": [ 7, 0, 7 ],
+            "to": [ 9, 16, 9 ],
+            "faces": {
+                "down":  { "texture": "#all", "cullface": "down" },
+                "up":    { "texture": "#all", "cullface": "up" },
+                "north": { "texture": "#all", "cullface": "north" },
+                "south": { "texture": "#all", "cullface": "south" },
+                "west":  { "texture": "#all", "cullface": "west" },
+                "east":  { "texture": "#all", "cullface": "east" }
```
Comment (Gegy) [inline src/main/resources/assets/ltextras/models/block/papyrus_stem.json]: As these horizontal faces don't reach the edge, I think they shouldn't have the cullface property :)

File: src/main/java/com/lovetropics/extras/block/PapyrusUmbelBlock.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    @Override
+    protected void randomTick(final BlockState state, final ServerLevel level, final BlockPos pos, final RandomSource random) {
+        //10% chance to stop ageing
+        if (random.nextInt(10) == 0) {
+            level.setBlockAndUpdate(pos, state.setValue(AGEING, false));
+        }
+
+        //10% chance to age
+        if (state.getValue(AGEING) && random.nextInt(10) == 0) {
```
Comment (Gegy) [inline src/main/java/com/lovetropics/extras/block/PapyrusUmbelBlock.java]: Nit: `getValue(AGING)` will always be true by this stage :)

File: src/main/java/com/lovetropics/extras/block/PapyrusStemBlock.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        return canGrowOn(stateBelow);
+    }
+
+    @Override
+    protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> builder) {
+        builder.add(TYPE);
+    }
+
+    private boolean canGrowOn(BlockState state) {
+        return GROWS_ON.contains(state.getBlock()) || state.is(ExtraBlocks.PAPYRUS_STEM);
```
Comment (Gegy) [inline src/main/java/com/lovetropics/extras/block/PapyrusStemBlock.java]: We might be able to use `BlockTags.DIRT` which also includes grass blocks and some other dirt-like blocks :)

File: src/main/java/com/lovetropics/extras/block/PapyrusStemBlock.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        return false;
+    }
+
+    @Override
+    public void performBonemeal(final ServerLevel serverLevel, final RandomSource randomSource, final BlockPos blockPos, final BlockState blockState) {
+
+    }
+
+    @Override
+    protected boolean canSurvive(final BlockState state, final LevelReader level, final BlockPos pos) {
```
Comment (Gegy) [inline src/main/java/com/lovetropics/extras/block/PapyrusStemBlock.java]: To ensure that the block gets removed when no longer in a valid position, I think we need to override `updateShape` and implement it there. 

`BambooStalkBlock.updateShape/tick` might be a nice example - it also has the fun 1 tick delay between each block breaking which is a fun effect :D

File: src/main/java/com/lovetropics/extras/block/PapyrusUmbelBlock.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.BooleanProperty;
+import net.minecraft.world.level.block.state.properties.EnumProperty;
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.VoxelShape;
+
+public final class PapyrusUmbelBlock extends Block {
```
Comment (Gegy) [inline src/main/java/com/lovetropics/extras/block/PapyrusUmbelBlock.java:20]: Not critical either, but might make sense for this one also to have `canSurvive`/`updateShape` logic to ensure it breaks without support? :)

**Review by Gegy — COMMENTED** [review]
This looks awesome, nice!

File: src/main/resources/assets/ltextras/models/block/papyrus_stem.json (hunk 1 above)
Comment (itsbazke) [reply src/main/resources/assets/ltextras/models/block/papyrus_stem.json]: doh, of course!

File: src/main/java/com/lovetropics/extras/block/PapyrusUmbelBlock.java (hunk 2 above)
Comment (itsbazke) [reply src/main/java/com/lovetropics/extras/block/PapyrusUmbelBlock.java]: ah good catch. i threw in the check to stop randomly ticking after.

Do you have any suggestion on how to stop the stem from randomly ticking, depending on the block above? and can something start randomly ticking again, after having returned false once? Thinking that it's unnecessary to do all the checks for the stem in randomTick unless the block above is air. But if it becomes air again it should start again

edit: dont really have a way to know about other blocks without changing a loooot :(

File: src/main/java/com/lovetropics/extras/block/PapyrusStemBlock.java (hunk 3 above)
Comment (itsbazke) [reply src/main/java/com/lovetropics/extras/block/PapyrusStemBlock.java]: yes, nice!

File: src/main/java/com/lovetropics/extras/block/PapyrusStemBlock.java (hunk 4 above)
Comment (itsbazke) [reply src/main/java/com/lovetropics/extras/block/PapyrusStemBlock.java]: Yes! Now they fall apart nicely. I just have to fix the sound

File: src/main/java/com/lovetropics/extras/block/PapyrusUmbelBlock.java (hunk 5 above)
Comment (itsbazke) [reply src/main/java/com/lovetropics/extras/block/PapyrusUmbelBlock.java:20]: Done and done!

**Review by Gegy — APPROVED** [review]
Really nice!

**Pull request merged**